cfg/config_util.go

```diff
@@ -55,3 +56,14 @@ func ListCacheTTLSecsToDuration(secs int64) time.Duration {
 func IsMetricsEnabled(c *MetricsConfig) bool {
 	return c.CloudMetricsExportIntervalSecs > 0 || c.PrometheusPort > 0
 }
+
+// IsGKEEnvironment returns true if the file descriptor mount option is set.
+func IsGKEEnvironment(mountConfig *Config) bool {
```

Comment:
The mountPoint string [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/acdf037408d7969cd398e1481055fae0cdad19c5/cmd/legacy_main.go#L417C40-L417C50) would have prefix /dev/fd/ in GKE environment. See [this](https://github.com/jacobsa/fuse/blob/b8f47b05858b6d8b6ba83df84ce26a7e13442663/unmount_linux.go#L13) for reference.

Reply:
Thanks @kislaykishore  for sharing the code links. Does this mean that the prefix "/dev/fd/" in question cannot be found in the fd mount options `mountConfig.FileSystem.FuseOptions`?

I looked into the linked files. Is the mountPoint string stored in any config struct? What would be the best way to access the `mountPoint` string for this check?

Reply:
No, it's not.  Perhaps you can plumb this information using the [createStorageHandle](https://github.com/GoogleCloudPlatform/gcsfuse/blob/980194f078497ef45c7b4a3b2922070ed653c648/cmd/legacy_main.go#L176C24-L176C43) method. Alternatively, see if you could use gcp Resource detector for this.

Reply:
Sounds good, plumbed through the createStorageHandle method and added a flag guard